### ISSUE TITLE
[FEATURE REQUEST] Some UI options

### ISSUE THREAD

**Issue by straycosmicllama** [jrpie/Launcher#41]
Some feature requests that might make this app more customisable
- Option to toggle icon visibility
- Adjust padding, alignment, font-size of elements etc
- Option to display  battery, weather
- Option to change date format possibly using momentjs format
- Notification dots
- Maybe material 3 buttons and switches?

Thanks for your work

**Comment by jrpie** [comment]
Thank you for the suggestions!

> Option to toggle icon visibility

I just opened #42 for this. Should be quite straight forward, I'll probably implement this very soon. (This week I don't really have time to work on this project).


> Adjust padding, alignment, font-size of elements etc

Being able to change font size sounds like a good idea. I wouldn't like to add very granular settings for this as that would make the settings quite cluttered.
I guess a setting for the clock would be quite nice (just added that to #12 ). The rest of the app should already follow the global font size setting, so I don't really see why there should be an additional setting for this.

What paddings and alignments would you like to change? I guess this is mostly about "All Apps"?
Having the app names centered might look quite nice (when hiding app icons, this should perhaps be the default?)

> Option to display battery, weather

How and where would you want the battery status to be displayed? When adding support for this, there should probably also be support network connection status, CPU load, memory, etc., right? This would be quite a large feature. I'm not quite about the best way to do this (where should all of that be shown? what settings are needed for this?) and whether or not I'd want to implement this at all.

I really don't want to add weather, as this would require sending the users location to some weather provider. I don't like the privacy implications of that. I think µLauncher should not open network connections at all.
I'm thinking aboud adding support for [widgeds](https://developer.android.com/develop/ui/views/appwidgets/host). That way, users would be able to chose a weather provider themselves.
(Battery status indicators might also be added this way making the suggestion above redundant).

Widgets are more of a long term idea, I probably won't work on that very soon.
For now you could use an app like [Shader Editor](https://github.com/markusfisch/ShaderEditor) to create a custom live wallpaper that depends on battery status.


> Option to change date format possibly using momentjs format

What date would you like to use? I thought the choice between ISO 8601 and the localized format should be sufficient. Do you want to use a format from a different locale?
I might add a setting to let the user set a custom format string for the [SimpleDateFormat](https://developer.android.com/reference/android/icu/text/SimpleDateFormat). This would require a lot of explanation though.

> Notification dots

yes that is a good idea. I think those should just be enabled by default, is there really a need to add an option for this?
I opened #43 for this.

> Maybe material 3 buttons and switches?

Personally I don't like Material 3 too much. However this seems to be a prerequisite for #31 anyway, so I'll probably add it at some time. Material 3 should probably be the default for devices supporting it.





What do you think about this?

**Comment by straycosmicllama** [comment]
Thanks for the response

> What paddings and alignments would you like to change?
Adjusting padding between elements of app list
App list alignment as right, center or left
> Option to display battery, weather
I have suggested this feature with reference from few minimal launcher in playstore
An example would be `indistractable launcher`
![image](https://github.com/user-attachments/assets/5579dc26-72bd-44d6-8b7f-3ab868462b67)thin
> Do you want to use a format from a different locale?
I think custom string format might be nice
> Notification dots
By notification dots I meant dots beside the app name when there is no app icon

**Comment by jrpie** [comment]
> Adjusting padding between elements of app list

Is that really necessary? With the app icons shown, the rows don't fit a lot closer together. Without icons the rows should of course be closer together, but this will happen naturally. So I think this should just depend on whether or not the icons are shown.

For an even more compact view I'd rather add a grid layout.

> App list alignment as right, center or left

👍🏼 

> indistractable launcher

That does indeed look nice. Added it to #12 

> I think custom string format might be nice

Also added to #12 

> By notification dots I meant dots beside the app name when there is no app icon

I see. That also sounds like a good idea!

**Comment by jstride** [comment]
The ability to add move the date/time to the top or bottom, left/right/centre would be nice. My use case is I have my family photos as my background that change randomly but the top or bottom is generally 'free' space. Often the date is in the middle of somebody's face at the moment

**Comment by jrpie** [comment]
Sounds like a good idea!

What layout would you prefer:
(a) top, centered
![image](https://github.com/user-attachments/assets/51367616-7608-4223-ab26-02b57faeb798)

(b) corner
![image](https://github.com/user-attachments/assets/82f8a60b-f98d-4846-88d5-050cae56a08d)

(c) corner, single line, ISO8601
![image](https://github.com/user-attachments/assets/39ab9727-d68f-4b2b-8720-ff4d126dee17)

(d) corner, stacked
![image](https://github.com/user-attachments/assets/946d0125-12af-49d7-94ea-fdfe9f809822)

...

I'm not really sure how to implement this best. I see two options:

1. Give the user a list of clock styles to choose from. A style would then be something like "top left corner, stacked"
2. Split the screen into serveral regions. Then the clock can be assigned to any of those region. This would interact nicely with #44 . Styles would be a separate setting independent of the position.

**Comment by jrpie** [comment]
I think I am in favor of 2., however this adds quite a lot of complexity and might be confusing

**Comment by jstride** [comment]
Option 2 seems more logical and would decouple the position from what the clock/date looks like

**Comment by jrpie** [comment]
This issue seems to be digressing into a more general discussion. I think a chat might be a better option for that, so I created groups on [matrix](https://s.jrpie.de/launcher-matrix) resp. [discord](https://s.jrpie.de/launcher-discord) (the two are bridged together so everyone can use whatever they like).

Since all the suggestions I want to implement have been moved to separate issues, I will close this one now.
